WordPress/gutenberg
# Components: Ship TypeScript types

## What?

Related to #35744

Properly ship first-party TypeScript declarations with the `@wordpress/components` package.

## Why?

We were relying on `@types`, which could be wrong or outdated.

**Follow-up: remove types from Definitely Typed**

## Testing Instructions

1. `npm run build`
2. In your IDE, go to component usages in packages other than wp-components. Check that IntelliSense and source mapping (e.g. "Go to Definition" in VS Code) works as expected.
